project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module

```diff
@@ -285,7 +285,8 @@ function c4m_user_get_initials($name) {
 function c4m_user_name($user) {
   $wrapper = entity_metadata_wrapper('user', $user);
 
-  if ($user->uid == 0) {
+  // When an user is created, this function is called without uid.
+  if (!isset($user->uid) || $user->uid == 0) {
```

Comment:
Just tested it with a first and second name and still fails. That check of the UID happens before checking the first name or the second name.